src/hooks/usePortflioPerformanceData/utils/calculateDailyPortfolioValue.ts

```diff
@@ -0,0 +1,49 @@
+import { HistoricalPriceData } from '@/types/historicalPrices';
+import { DailyPortfolio } from '../../../utils/portfolioCalculations/getDailyPortfolio';
+import { CurrentPortfolioItem } from '@/types/currentPortfolio';
+import { getLatestAvailablePrices } from './getLatestAvailablePrice';
+import { getApplicablePrice } from './getApplicablePrice';
+
+export const calculateDailyPortfolioValues = (
+  dailyPortfolio: DailyPortfolio,
+  stockPrices: HistoricalPriceData[],
+): { date: string; portfolio_value: number }[] => {
+  let lastAvailablePrices = getLatestAvailablePrices(stockPrices);
```

Comment:
This function is hard to understand, ideally it should be deconstructed to smaller functions and more deterministic steps. You use `let` keywords for two different variables that makes it much harder to follow what exactly is happening. 
Do you think you can come up with a refactored version of this function that will be easier to understand.